vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java

```diff
@@ -356,5 +363,32 @@ private Sort getSortingParameters(String sortedBy, String order) {
         }
         return sortingParameters;
     }
+
+    @Override
+    public void updateNameAndDescription(ISpace space, SpaceForm spaceForm) {
+        space.setName(spaceForm.getDefaultName().getText());
+        space.setDescription(spaceForm.getDefaultDescription().getText());
+        List<ILocalizedText> names = space.getSpaceNames();
+        List<ILocalizedText> descriptions = space.getSpaceDescriptions();
+        
+        addSpaceDetails(space,spaceForm.getDefaultName(), names);
+        addSpaceDetails(space,spaceForm.getDefaultDescription(), descriptions);
+        for(LocalizedTextForm title:spaceForm.getNames()) {   
+            addSpaceDetails(space,title, names);
+        }
+        for(LocalizedTextForm text:spaceForm.getDescriptions()) {
+            addSpaceDetails(space,text, descriptions);
+        }       
+    }
+    
+    /**
+     * Adds name to spaceNames List of the given space object.
+     * @param space The space to which the name will be added.
+     * @param name The localized text form containing the name to be added.
+     */
+    @Override
+    public void addSpaceDetails(ISpace space, LocalizedTextForm name, List<ILocalizedText> localizedTextList) {
+        localizedTextFactory.createLocalizedText(space, name, localizedTextList);        
```

Comment:
same as before, this creates not updates, right? so what happens to any existing localized text?